fdb-extensions/src/main/java/com/apple/foundationdb/async/RankedSet.java

```diff
@@ -246,6 +382,80 @@ public RankedSet(Subspace subspace, Executor executor) {
         });
     }
 
+    /**
+     * Removes a key from the set.
+     * @param tc the transaction to use to access the database
+     * @param key the key to remove
+     * @return a future that completes to {@code true} if the set was modified, that is, if the key was present before this operation
+     */
+    public CompletableFuture<Boolean> remove(TransactionContext tc, byte[] key) {
+        checkKey(key);
+        return tc.runAsync(tr ->
+                countCheckedKey(tr, key)
+                        .thenCompose(count -> {
+                            if (count == null || count <= 0) {
+                                return READY_FALSE;
+                            }
+                            // This works even if the current set does not track duplicates but duplicates were added
+                            // earlier by one that did.
+                            final boolean duplicate = count > 1;
+                            final int nlevels = config.getNLevels();
+                            final List<CompletableFuture<Void>> futures = new ArrayList<>(nlevels);
+                            for (int li = 0; li < nlevels; ++li) {
+                                final int level = li;
+
+                                final CompletableFuture<Void> future;
+
+                                if (duplicate) {
+                                    // Always subtract one, never clearing a key. It is possible for this to leave
+                                    // a key with a count of zero, if duplicates were inserted with different hash functions.
+                                    Function<byte[], Void> decrement = k -> {
+                                        tr.mutate(MutationType.ADD, subspace.pack(Tuple.from(level, k)), encodeLong(-1));
+                                        return null;
+                                    };
+                                    if (level == 0) {
+                                        decrement.apply(key);
+                                        future = DONE;
+                                    } else {
+                                        future = getPreviousKey(tr, level, key, true).thenApply(decrement);
+                                    }
+                                } else {
+                                    // This could be optimized to check the hash for which levels should have this key.
+                                    // That would require that the hash function never changes, though.
+                                    // This allows for it to change, with the distribution perhaps getting a little uneven
+                                    // as a result. It even allows for the hash function to return a random number.
```

Comment:
Is it worth having a `RANDOM_HASH_FUNCTION`? Roughly, `threadLocalRandom.getInt()`. And a test to show that it works?

Reply:
I added the basic test for this.